modules/aggs-matrix-stats/src/test/java/org/opensearch/search/aggregations/matrix/stats/MatrixStatsAggregatorTests.java

```diff
@@ -105,10 +105,10 @@ public void testTwoFields() throws Exception {
             for (int docId = 0; docId < numDocs; docId++) {
                 Document document = new Document();
                 fieldAValues[docId] = randomDouble();
-                document.add(new SortedNumericDocValuesField(fieldA, NumericUtils.doubleToSortableLong(fieldAValues[docId])));
+                document.add(SortedNumericDocValuesField.indexedField(fieldA, NumericUtils.doubleToSortableLong(fieldAValues[docId])));
```

Comment:
1. Lets not enable by default ? One option is to introduce a new parameter for skipper similar to docValues / index / stored_fields to the user.  [ Of course docValues must be true for skipper to be true ]

2. Also we must see if we need to enable in experimental mode for now until search takes advantage of this data structure.

3. Do we have any benchmarks in terms of storage if we enable for all fields like this ? We can try atleast in few OSB workloads with and without this change.